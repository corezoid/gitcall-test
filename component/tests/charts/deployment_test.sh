#!/usr/bin/env bash
$CZ helm template | sed -n '/deployment\.yaml/,/Source\:/p'
