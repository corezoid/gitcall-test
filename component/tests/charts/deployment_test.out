# Source: gitcall_release/charts/gitcall/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: RELEASE-NAME-gitcall-deployment
  labels:
    release: "RELEASE-NAME"
    app: "gitcall"
    tier: "gitcall"
spec:
  selector:
    matchLabels:
      release: "RELEASE-NAME"
      app: "gitcall"
      tier: "gitcall"
  strategy:
    type: RollingUpdate
  replicas: 1
  template:
    metadata:
      annotations:
        checksum/config: a08cfc4022a5954027e1378a3599b92d2bc9daafd28f7ebb0171a71f1b4b457c
        checksum/env: ec15e7084e8782cb71ce567f5bc5bd5798b393e5afca88de52f6a39a0157ee2e
        prometheus.io/path: /metrics
        prometheus.io/port: "9100"
        prometheus.io/scrape: "true"
      labels:
        release: "RELEASE-NAME"
        app: "gitcall"
        tier: "gitcall"
    spec:
      terminationGracePeriodSeconds: 40
      initContainers:
        - name: init-wait-postgresql
          image: alpine
          command: ["sh", "-c", "until nc -zvw1 ${GITCALL_POSTGRESQL_HOST} ${GITCALL_POSTGRESQL_PORT}; do echo waiting for dependencies; sleep 2; done;"]
          env:
            # todo secretalreadyexist true            
            - name: GITCALL_POSTGRESQL_HOST
              value: "postgresql.host"
            - name: GITCALL_POSTGRESQL_PORT
              value: "5432"
            - name: GITCALL_POSTGRESQL_USERNAME
              value: "postgres"
            - name: GITCALL_POSTGRESQL_PASSWORD
              value: "pass"
            - name: GITCALL_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: GITCALL_RABBITMQ_PORT
              value: "5672"
            - name: GITCALL_RABBITMQ_USERNAME
              value: "guest"
            - name: GITCALL_RABBITMQ_PASSWORD
              value: "guest"
            - name: GITCALL_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_TASK_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_TASK_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_VHOST
              value: "/"
        - name: init-wait-rabbitmq
          image: alpine
          command: ["sh", "-c", "until nc -zvw1 ${GITCALL_RABBITMQ_HOST} ${GITCALL_RABBITMQ_PORT}; do echo waiting for dependencies; sleep 2; done;"]
          env:
            # todo secretalreadyexist true            
            - name: GITCALL_POSTGRESQL_HOST
              value: "postgresql.host"
            - name: GITCALL_POSTGRESQL_PORT
              value: "5432"
            - name: GITCALL_POSTGRESQL_USERNAME
              value: "postgres"
            - name: GITCALL_POSTGRESQL_PASSWORD
              value: "pass"
            - name: GITCALL_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: GITCALL_RABBITMQ_PORT
              value: "5672"
            - name: GITCALL_RABBITMQ_USERNAME
              value: "guest"
            - name: GITCALL_RABBITMQ_PASSWORD
              value: "guest"
            - name: GITCALL_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_TASK_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_TASK_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_VHOST
              value: "/"
      containers:
        - name: main
          imagePullPolicy: "Always"
          image: docker-hub.middleware.biz/gitcall/gitcall:aVersion
          command: ["/gitcall/gitcall"]
          args: ["--config", "/gitcall/gitcall.json"]
          env:
            - name: GOPS_CONFIG_DIR
              value: /tmp            
            - name: GITCALL_POSTGRESQL_HOST
              value: "postgresql.host"
            - name: GITCALL_POSTGRESQL_PORT
              value: "5432"
            - name: GITCALL_POSTGRESQL_USERNAME
              value: "postgres"
            - name: GITCALL_POSTGRESQL_PASSWORD
              value: "pass"
            - name: GITCALL_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: GITCALL_RABBITMQ_PORT
              value: "5672"
            - name: GITCALL_RABBITMQ_USERNAME
              value: "guest"
            - name: GITCALL_RABBITMQ_PASSWORD
              value: "guest"
            - name: GITCALL_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_TASK_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_TASK_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_TASK_RABBITMQ_VHOST
              value: "/"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_HOST
              value: "rabbitmq.host"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PORT
              value: "5672"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_USERNAME
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_PASSWORD
              value: "guest"
            - name: DUNDERGITCALL_SYSTEM_RABBITMQ_VHOST
              value: "/"          
          resources:
            limits:
              cpu: "100m"
              memory: "200Mi"
            requests:
              cpu: "300m"
              memory: "400Mi"
          volumeMounts:
            - name: config
              mountPath: /gitcall/gitcall.json
              subPath: gitcall.json
            - name: tmp
              mountPath: /tmp
          securityContext:
            runAsUser: 501
            runAsGroup: 501
            runAsNonRoot: true
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
            privileged: false
            capabilities:
              drop:
                - NET_ADMIN
                - SYS_ADMIN
                - MAC_ADMIN
                - SETUID
                - SETGID
                - FOWNER
                - CHOWN
                - MKNOD
      volumes:
        - name: config
          configMap:
            name: RELEASE-NAME-gitcall-config
        - name: tmp
          emptyDir: {}
---
# Source: gitcall_release/charts/gitcall/templates/autoscale.yaml
